app/javascript/components/HomePageContent.js

```diff
@@ -4,19 +4,26 @@ import ResultsPanel from './ResultsPanel'
 import StudySearchProvider from 'components/search/StudySearchProvider'
 import SearchFacetProvider from 'components/search/SearchFacetProvider'
 import UserProvider from 'components/UserProvider'
+import ErrorBoundary from 'lib/ErrorBoundary'
 
 /**
  * Wrapper component for search and result panels
  */
 export default function HomePageContent() {
   return (
-    <UserProvider>
-      <SearchFacetProvider>
-        <StudySearchProvider>
-          <SearchPanel/>
-          <ResultsPanel/>
-        </StudySearchProvider>
-      </SearchFacetProvider>
-    </UserProvider>
+    <ErrorBoundary>
```

Comment:
Needing to wrap each rendered component in `ErrorBoundary` seems tedious and error-prone -- I could see it being overlooked.

Could this boilerplate go away only if/when hooks support `componentDidCatch`, or could it be refactored beforehand?

Reply:
so this is an interesting question that I'm not sure I know the best path.  I've been operating under the assumption that it's not worth wrapping every component in an error boundary, but rather we should only wrap functional areas.  But I could see either approach being correct.  I've got a higher-order component to help with boilerplate, but you're right that a hook would be easier